### PR TITLE
[profiler] add profiler parsing support for custom device. 

### PR DESCRIPTION
**Pull request by MooYeh279**
We hope PyTorch profiling parsing ability can also be applicable to custom devices. Based on previous  work  https://github.com/pytorch/pytorch/pull/101554, we have made supplementary updates to PyTorch profiling to extend its parsing capabilities for custom devices. These modifications do not affect the original logic of the code and mainly include the following aspects:
1. Added the relevant logic for use_device in torch.profiler.profiler._KinetoProfile.
2. In torch.autograd.profiler and torch.autograd.profiler_util, custom devices profiling data  parsing ability has been added using privateuse1 and use_device attributes.
3. In torch._C._autograd.pyi and torch._C._autograd.pyi, custom devices related attributes have been added. The underlying C++ 
logic will be added in subsequent pull requests.

### CODE REVIEW

**Comment by MooYeh279** [comment]
release notes: profiler

**Comment by MooYeh279** [comment]
@aaronenyeshi sorry to bother. why my pr have no responding? (eg. ther test cases are not triggered) Thanks!

**Comment by aaronenyeshi** [comment]
> @aaronenyeshi sorry to bother. why my pr have no responding? (eg. ther test cases are not triggered) Thanks!

I've triggered the tests to run. Please take a look at the broken tests please.

**Comment by MooYeh279** [comment]
> 
@aaronenyeshi Thanks, I have fix the bug. What should I do next?

**Comment by MooYeh279** [comment]
@aaronenyeshi How can trigger the next workflow? (eg. lintrunner. My code did not pass the check before, I have modified follow the suggestions.) Thanks!

**Comment by MooYeh279** [comment]
@aaronenyeshi Hi, can this work be reviewed? (there is an unrelated failure)

**Comment by MooYeh279** [comment]
@aaronenyeshi Hi, sorry to bother, there is an unrelated failure, can you help to re-run this check?
image

**Comment by MooYeh279** [comment]
@aaronenyeshi Hi, I have re-run the check. Other workflows need approval.

**Comment by MooYeh279** [comment]
@aaronenyeshi @soulitzer Hi, how can this code be reviewed?

File: torch/autograd/profiler.py [hunk 1]
```diff
@@ -221,6 +223,14 @@ def __init__(
                 use_kineto
             ), "Device-only events supported only with Kineto (use_kineto=True)"
 
+        if self.use_device and self.use_device != _get_privateuse1_backend_name():
+            warn(f"{self.use_device} doesn't support profile.")
+            self.use_device = None
+
+        if self.use_device == "cuda":
```
Comment (aaronenyeshi) [inline torch/autograd/profiler.py:230]: If use_device is cuda, won't it get set to none on line 228?

**Review by aaronenyeshi — CHANGES_REQUESTED** [review]
Overall looks good, just a quick question, and we can land this.

File: torch/autograd/profiler.py (hunk 1 above)
Comment (MooYeh279) [reply torch/autograd/profiler.py:230]: Thanks for your opinion. if use_device is cuda, use_device indeed will be set to none! I will fix it now!

**Comment by MooYeh279** [comment]
@aaronenyeshi Hi, can you help me trigger the other workflows? Thanks!

**Review by aaronenyeshi — APPROVED** [review]
Thanks for fixing to my comments. This LGTM.

**Pull request closed without merging**